packages/react-query/src/shared/proxy/useQueriesProxy.ts

```diff
@@ -0,0 +1,83 @@
+import { QueryOptions } from '@tanstack/react-query';
+import { TRPCClient, TRPCClientError } from '@trpc/client';
+import {
+  AnyProcedure,
+  AnyQueryProcedure,
+  AnyRouter,
+  Filter,
+  inferProcedureInput,
+} from '@trpc/server';
+import {
+  createRecursiveProxy,
+  inferTransformedProcedureOutput,
+} from '@trpc/server/shared';
+import { getQueryKey } from '../../internals/getQueryKey';
+import { TrpcQueryOptionsForUseQueries } from '../../internals/useQueries';
+
+type GetQueryOptions<
+  TRouter extends AnyRouter,
+  TProcedure extends AnyProcedure,
+  TPath extends string,
+> = <TData = inferTransformedProcedureOutput<TProcedure>>(
+  input: inferProcedureInput<TProcedure>,
+  opts?: TrpcQueryOptionsForUseQueries<
+    TPath,
+    inferProcedureInput<TProcedure>,
+    inferTransformedProcedureOutput<TProcedure>,
+    TData,
+    TRPCClientError<TRouter>
+  >,
+) => TrpcQueryOptionsForUseQueries<
+  TPath,
+  inferProcedureInput<TProcedure>,
+  inferTransformedProcedureOutput<TProcedure>,
+  TData,
+  TRPCClientError<TRouter>
+>;
+
+/**
+ * @internal
+ */
+export type UseQueriesProcedureRecord<
+  TRouter extends AnyRouter,
+  TPath extends string = '',
+> = {
+  [TKey in keyof Filter<
+    TRouter['_def']['record'],
+    AnyRouter | AnyQueryProcedure
+  >]: TRouter['_def']['record'][TKey] extends AnyRouter
+    ? UseQueriesProcedureRecord<
+        TRouter['_def']['record'][TKey],
+        `${TPath}${TKey & string}.`
+      >
+    : GetQueryOptions<
+        TRouter,
+        TRouter['_def']['record'][TKey],
+        `${TPath}${TKey & string}`
+      >;
+};
+
+/**
+ * Create proxy for `useQueries` options
+ * @internal
+ */
+export function createUseQueriesProxy<TRouter extends AnyRouter>(
+  client: TRPCClient<TRouter>,
+) {
+  return createRecursiveProxy((opts) => {
+    const path = opts.path.join('.');
+    const [input, ...rest] = opts.args;
+
+    const queryKey = getQueryKey(path, input);
```

Comment:
On further comment, that could enable using `getArrayQueryKey` here instead?

Reply:
yeahh we should do this instead

Reply:
not possible since `prefetchQuery` converts to array query key internally 😅